### ISSUE TITLE
Separate build and debug remotes

### ISSUE THREAD

**Issue by tksuoran** [microsoft/vscode#150721]
Current Remote SSH, Remote WSL and Remote Docker extensions are great, but it appears the same remote is used for building and debugging. This limits how VSCode can be used for cross-compilation development. It would be great if it would be possible to have two separate remotes simultaneously: One remote used for building, and a second remote for debugging. This would allow for example to use a docker remote for standardized environment for cross-compilation, while using ssh remote for debugging.

Thanks!

**Comment by TomTravis** [comment]
yes using containers (chroots), do you need to share resources (FS)
Dynamic users? who manages this? which system?

**Comment by roblourens** [comment]
How are we supposed to deploy your project from one remote to another? 

This isn't really a good match for our remote model. You might want to combine one remote for building with remote debugging (not "vscode remote" but starting the app in debug mode with a port open and configuring your launch.json to attach to that IP and port). Or two remote windows, managing them yourself.
